sdk/core/System.ClientModel/src/Message/PipelineMessage.cs

```diff
@@ -146,7 +146,7 @@ public void Apply(RequestOptions options)
     /// <param name="key">The key for the property in the message's property
     /// bag.</param>
     /// <param name="value">The value of the property.</param>
-    public void SetProperty(Type key, object value) =>
+    public void SetProperty(Type key, object? value) =>
```

Comment:
If we want to be able to set properties on PipelineMessage to `null` we'll need to make this nullable.  The changes to ArrayBackedPropertyBag accommodate this change.

If we decide to keep this API change, I'll add it to https://github.com/Azure/azure-sdk-for-net/pull/42328.

Reply:
What's the scenario where we'd want to set properties to null?   
_(I don't have an opinion, I'm just curious about the end-to-end)_

Reply:
Answered elsewhere - to un-set properties without adding another member.